src/daemon/daemon.cpp

```diff
@@ -170,25 +171,30 @@ bool t_daemon::run(bool interactive)
     cryptonote::rpc::DaemonHandler rpc_daemon_handler(mp_internals->core.get(), mp_internals->p2p.get());
     cryptonote::rpc::ZmqServer zmq_server(rpc_daemon_handler);
 
-    if (!zmq_server.addTCPSocket(zmq_rpc_bind_address, zmq_rpc_bind_port))
+    if (!zmq_rpc_disabled)
     {
-      LOG_ERROR(std::string("Failed to add TCP Socket (") + zmq_rpc_bind_address
-          + ":" + zmq_rpc_bind_port + ") to ZMQ RPC Server");
+      if (!zmq_server.addTCPSocket(zmq_rpc_bind_address, zmq_rpc_bind_port))
+      {
+        LOG_ERROR(std::string("Failed to add TCP Socket (") + zmq_rpc_bind_address
+            + ":" + zmq_rpc_bind_port + ") to ZMQ RPC Server");
 
-      if (rpc_commands)
-        rpc_commands->stop_handling();
+        if (rpc_commands)
+          rpc_commands->stop_handling();
 
-      for(auto& rpc : mp_internals->rpcs)
-        rpc->stop();
+        for(auto& rpc : mp_internals->rpcs)
+          rpc->stop();
 
-      return false;
-    }
+        return false;
+      }
```

Comment:
That looks like a very diff-spammy way to say: if (!zmq_rpc_disabled && !zmq_server.addTcpSocket(...))

Reply:
Not really, and I nearly did what you are suggesting. The addTcpSocket failure (if check) is independent to deciding to disable zmq.

Reply:
I can rejig it if you are really bugged by it, but then of course I also need to wrap zmq_server.run(); in an if block.

Reply:
OK, there's a message that's extra. Fine :)